src/PrestaShopBundle/Security/Admin/TokenizedControllerListener.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Security\Admin;
+
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Csrf\CsrfTokenManager;
+
+/**
+ * Class TokenizedControllerListener: A middleware to check CSRF token value in POST HTTP requests.
+ *
+ * The intent/tokenID is the sf route name of the action.
+ * A POST request is checked only if the controller implements TokenizedController
+ *
+ * In case of error, a warning is triggered for now. No blocking behavior for ALPHA, but we must make it blocking after 1.7.0.
+ * FIXME: >1.7.0: this should be blocking
+ *
+ * @package PrestaShopBundle\Security\Admin
+ */
+class TokenizedControllerListener
+{
+    /**
+     * @var CsrfTokenManager
+     */
+    private $csrfProvider;
+
+    public function __construct(CsrfTokenManager $csrfProvider)
```

Comment:
be careful, $csrfProvider class do exists, and doesnt have the same use.

Let's call it $csrfTokenManager :)